---
title: "Preprocessed Adult Height GWAS Summary Data"
author: Xiang Zhu
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

[zenodo-height2014]: https://doi.org/10.5281/zenodo.1443565
[Zhu and Stephens (2017)]: https://projecteuclid.org/euclid.aoas/1507168840
[Wen and Stephens (2010)]: https://projecteuclid.org/euclid.aoas/1287409368 
[`get_corr.m`]: https://github.com/stephenslab/rss/blob/master/misc/get_corr.m
[`rss/src`]: https://github.com/stephenslab/rss/tree/master/src
[`rss/src_vb`]: https://github.com/stephenslab/rss/tree/master/src_vb
[Wood et al. (2014)]: https://www.ncbi.nlm.nih.gov/pubmed/25282103
[1000 Genomes Project Phase 1]: https://www.ncbi.nlm.nih.gov/pubmed/20981092
[HapMap Release 24]: https://www.ncbi.nlm.nih.gov/pubmed/17943122

This page provides information on the preprocessed adult height
genome-wide association study (GWAS) summary statistics and
estimated linkage disequilibrium (LD) matrices, which were
created and analyzed in the following publication.

> Zhu, Xiang; Stephens, Matthew. Bayesian large-scale multiple regression with summary statistics from genome-wide association studies. Ann. Appl. Stat. 11 (2017), no. 3, 1561--1592. [DOI:10.1214/17-AOAS1046](https://projecteuclid.org/euclid.aoas/1507168840).

This dataset is publicly available at <https://doi.org/10.5281/zenodo.1443565>,
and can be referenced in a journal's "Data availability" section as
[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.1443565.svg)][zenodo-height2014]

If you find this dataset useful in your research,
please kindly cite the publication listed above, [Zhu and Stephens (2017)][].

## GWAS summary statistics

The folder `mat_files` contains the GWAS summary statistics
for each of the 22 autosomes.

```matlab
>> load mat_files/height2014.chr22.mat;
>> whos
  Name             Size                Bytes  Class     Attributes

  H            15599x758            94592336  double
  Nsnp         15599x1                 62396  int32
  betahat      15599x1                124792  double
  chr          15599x1                 62396  int32
  cummap       15599x1                124792  double
  pos18        15599x1                 62396  int32
  pos19        15599x1                 62396  int32
  se           15599x1                124792  double
```

Most variables above are self-explanatory.
The single-SNP GWAS summary statistics `{betahat, se}` are published in [Wood et al. (2014)][].
The matrix `H` contains the phased haplotypes of 379 European ancestry individuals
in the [1000 Genomes Project Phase 1][].
The vector `cummap` contains the genetic map of
[HapMap Release 24][] European-ancestry population.
Note that `{betahat, se, H}` must use the SAME way of coding alleles;
otherwise RSS results will be severely distorted.  

## Estimated LD matrices

The folder `estimated_ld_sparse` contains the estimated LD matrices
for each of the 22 autosomes.
The estimation method is detailed in [Wen and Stephens (2010)][],
and is implemented by [`get_corr.m`][].

Files `R.chr*.mat` were generated by setting `cutoff=1e-8` in [`get_corr.m`][].
Files `R.chr*.3.mat` were generated by setting `cutoff=1e-3`.
These LD matrices are stored as sparse matrices to save space.

```matlab
>> load estimated_ld_sparse/R.chr22.mat;
>> whos R
  Name          Size                   Bytes  Class     Attributes

  R         15599x15599            592534032  double    sparse
```

Of note, if you run RSS MCMC programs ([`rss/src`][]) with these LD data,
please first convert them to full matrices: `R=full(R)`.
I recommend using full LD matrices in MCMC
because each iteration of MCMC involves multiple matrix indexing operations
(i.e. sampling SNPs to include in or exclude from the current regression model),
and based on my experiments, full matrix indexing is much faster than sparse matrix indexing
(at least in MATLAB).

However, if you run RSS VB programs ([`rss/src_vb`][]) with these LD data,
please do NOT convert them to full matrices.
Indeed, RSS VB programs require that input LD matrices must be sparse.

